test/scheduler_parameter_visibilities_test.py

```diff
@@ -0,0 +1,120 @@
+                c = luigi.Parameter(default="c", visibility=ParameterVisibility.PRIVATE)
+                d = luigi.Parameter(default="d", visibility=ParameterVisibility.PUBLIC)
+
+                def requires(self):
+                    return required_task
+
+                def run(self):
+                    yield dynamic_task
+
+            dynamic_task = DynamicTask()
+            required_task = RequiredTask()
+            task = Task()
+
+            w.add(task)
+            w.run()
+
+            time.sleep(1)
+            task_deps = s.dep_graph(task_id=task.task_id)
+            required_task_deps = s.dep_graph(task_id=required_task.task_id)
+            dynamic_task_deps = s.dep_graph(task_id=dynamic_task.task_id)
+
+            self.assertEqual('Task(a=a, d=d)', task_deps[task.task_id]['display_name'])
+            self.assertEqual('RequiredTask(required_public=required_param)',
+                             required_task_deps[required_task.task_id]['display_name'])
+            self.assertEqual('DynamicTask(dynamic_public=dynamic_public)',
+                             dynamic_task_deps[dynamic_task.task_id]['display_name'])
+
+            self.assertEqual({'a': 'a', 'd': 'd'}, task_deps[task.task_id]['params'])
+            self.assertEqual({'required_public': 'required_param'},
+                             required_task_deps[required_task.task_id]['params'])
+            self.assertEqual({'dynamic_public': 'dynamic_public'},
+                             dynamic_task_deps[dynamic_task.task_id]['params'])
+
+    def test_public_and_hidden_params(self):
+        s = luigi.scheduler.Scheduler(send_messages=True)
+        with luigi.worker.Worker(scheduler=s) as w:
+            class Task(RunOnceTask):
+                a = luigi.Parameter(default="a")
+                b = luigi.Parameter(default="b", visibility=ParameterVisibility.HIDDEN)
+                c = luigi.Parameter(default="c", visibility=ParameterVisibility.PRIVATE)
+                d = luigi.Parameter(default="d", visibility=ParameterVisibility.PUBLIC)
+
+            task = Task()
+
+            w.add(task)
+            w.run()
+
+            time.sleep(1)
+            t = s._state.get_task(task.task_id)
+            self.assertEqual({'b': 'b'}, t.hidden_params)
+            self.assertEqual({'a': 'a', 'd': 'd'}, t.public_params)
+            self.assertEqual({'a': 0, 'b': 1, 'd': 0}, t.param_visibilities)
+
+
+class Task(RunOnceTask):
+    a = luigi.Parameter(default="a")
+    b = luigi.Parameter(default="b", visibility=ParameterVisibility.HIDDEN)
+    c = luigi.Parameter(default="c", visibility=ParameterVisibility.PRIVATE)
+    d = luigi.Parameter(default="d", visibility=ParameterVisibility.PUBLIC)
+
+
+class RemoteSchedulerParameterVisibilitiesTest(server_test.ServerTestBase):
+    def test_public_params(self):
+        task = Task()
+        luigi.build(tasks=[task], workers=2, scheduler_port=self.get_http_port())
+
+        time.sleep(1)
```

Comment:
sleeping in tests is not ideal but okay. :)

Reply:
Proboaly, it's mistake :(
I wanted to add delay only for remote scheduler and only to guarantee that all messages were delivered, because i thought that there is a chance that message will not be delivered